src/main/groovy/io/openliberty/tools/gradle/tasks/AbstractServerTask.groovy

```diff
@@ -746,6 +827,10 @@ abstract class AbstractServerTask extends AbstractTask {
             combinedEnvProperties = properties
         }
 
+        return combinedEnvProperties;
```

Comment:
One more comment...the way this combined properties before was fine because it just used them locally to write the file. But you are potentially taking the result and calling combineServerEnvProperties again with it. So setting the combinedEnvProperties to either projectProperties or properties can have some interesting results later (with the original Properties objects getting altered). So let's create a new Properties object (at line 816) and do a putAll at lines 819 and 827 to be safe.